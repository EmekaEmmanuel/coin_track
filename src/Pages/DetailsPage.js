import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Details from '../Components/Details';
import Footer from '../Components/Footer';
import Nav from '../Components/Nav';

function DetailsPage() {
  const myCoins = useSelector((state) => state.coin.coins);
  const { id } = useParams();
  const coin = (myCoins.filter((coin) => coin.id === id))[0];

  return (
    <div className="app_container">
      <Nav />
      <Details coin={coin} />
      <Footer />
    </div>
  );
}

export default DetailsPage;
