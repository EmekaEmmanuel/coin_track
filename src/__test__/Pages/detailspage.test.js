import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import DetailsPage from '../../Pages/DetailsPage';

it('RoCoin Details page renders correctly', () => {
  const myCoins = store.getState().coin.coins;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <DetailsPage myCoins={myCoins} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
