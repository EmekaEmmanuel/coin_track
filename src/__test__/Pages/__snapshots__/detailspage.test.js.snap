// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page renders correctly 1`] = `
<div
  className="app_container"
>
  <div
    className="app_container"
  >
    <header
      className="home_header_box header_line"
    >
      <div
        className="home_header "
      >
        <div
          className="logo_box"
        >
          <figure>
            <img
              alt=""
              className="logo_img"
              src="crypto_image.png"
            />
          </figure>
          <h2
            className="nav_font1"
          >
            monie track
          </h2>
        </div>
        <nav
          className="nav_tag"
        >
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Track 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Happenings 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              className="nav_font2 td_none"
              href="/ChartPage"
              onClick={[Function]}
            >
              Chart 
            </a>
          </li>
        </nav>
        <div
          className="notify_box"
        >
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
        </div>
        <div
          className="hamburger_container"
        >
          <figure>
            <img
              alt="Hamburger_menu"
              src="hamburger.png"
            />
          </figure>
        </div>
      </div>
    </header>
  </div>
  <div
    className="app_container"
  >
    <main
      className="main_tag"
    >
      <section
        className="details_section1"
      >
        <p
          className="details_font1"
        >
          LIVE MARKET CHANGES
        </p>
        <h1
          className="details_font2"
        >
          Get real time
           
           
          details
        </h1>
        <p
          className="details_font3"
        >
          Your favorite place to see what's new.
        </p>
      </section>
    </main>
  </div>
  <div
    className="app_container"
  >
    <footer
      className="home_footer"
    >
      <article
        className="home_footer_article"
      >
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Company
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Trade
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Help
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Products
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
      </article>
      <div
        className="footer_copyright"
      >
        Â© 2022 monie track. All rights reserved.
      </div>
    </footer>
  </div>
</div>
`;
